### DISCUSSION TITLE
Published package shows up under "pip list" but will give a "ModuleNotFoundError" when imported

### DISCUSSION THREAD

**Discussion by nednoodlehead** [PyO3/maturin#1116] [Q&A] [answered]
I created a project, it is uploaded on Pypi under the name [punge-backend](https://pypi.org/project/punge-backend/). All of the code, cargo.toml, and pyproject.toml is [here](https://github.com/nednoodlehead/punge/tree/master/rust). 

This is most definitively user error, but I cannot find any part that is wrong.

-The module shows up on "pip list"
-Pip uninstall, then pip install does not fix the error
-The module's file structure seems to be regular, mimicking that of other, working packages.
![image](https://user-images.githubusercontent.com/102321690/191377549-9b91a1ea-93df-4b96-995f-a40478df70e2.png)
-I used "maturin publish" to publish the project to Pypi (if it wasn't obvious)

**Comment by messense** [comment, marked as answer]
https://pyo3.rs/v0.17.1/module.html

> The name of the module must match the name of the .so or .pyd file.

https://github.com/nednoodlehead/punge/blob/e4bcfdd59717ee4b55c27b119122250804b83b96/rust/src/lib.rs#L113-L114

This should be

```rust
#[pymodule]
fn punge_backend(_py: Python, m: &PyModule) -> PyResult<()> {
```

**Reply by nednoodlehead** [reply]
Yeah, sorry, I had just forgotten to update that. It does now line up with the module name.

**Reply by nednoodlehead** [reply]
The error still persists though @messense. Forgot to mention that.
